pkg/api/vlabs/validate.go

```diff
@@ -636,9 +636,10 @@ func (a *Properties) validateAddons() error {
 				}
 			}
 
-			switch addon.Name {
-			case "cluster-autoscaler":
-				if to.Bool(addon.Enabled) {
+			// Validation for addons if they are enabled
+			if to.Bool(addon.Enabled) {
```

Comment:
Wrapping this entire case/switch in a "is this addon enabled?" condition fixed a really nasty bug due to my adding aad addon validation here. Essentially, the validation below is (almost) entirely only valid in the context of the addon being enabled, so we also get to remove the `if to.Bool(addon.Enabled) {` for each addon case.

There was one "if addon is disabled" validation for `cloud-node-manager`, which was moved out to a new `else` block below.